Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -111,8 +111,9 @@ public async Task<ContestDetailsServiceModel> GetContestDetails(int id)
 
             var canShowProblemsInCompete = (!contest!.HasContestPassword && !contest.IsOnlineExam && contestActivityEntity.CanBeCompeted) || userIsAdminOrLecturerInContest;
             var canShowProblemsInPractice = (!contest.HasPracticePassword && contestActivityEntity.CanBePracticed) || userIsAdminOrLecturerInContest;
+            var cantShowProblemsInCompeteForAnonymous = user.Id == null && contestActivityEntity.CanBeCompeted;
 
-            if (!canShowProblemsInPractice && !canShowProblemsInCompete)
+            if ((!canShowProblemsInPractice && !canShowProblemsInCompete) || cantShowProblemsInCompeteForAnonymous)
```

Comment:
The other booleans are named as per the positive scenario and you're using "!", but this new variable is negative. It would be better if they all follow the same naming so its easier to read.

Reply:
Reworked.